src/main/java/org/mutabilitydetector/unittesting/matchers/reasons/ProvidedOtherClass.java

```diff
@@ -291,22 +288,11 @@ protected boolean matchesSafely(final MutableReasonDetail reasonDetail) {
             final CodeLocation<?> codeLocation = reasonDetail.codeLocation();
             if (reasonDetail.reason().isOneOf(MUTABLE_TYPE_TO_FIELD) && codeLocation instanceof FieldLocation) {
                 final FieldLocation location = (FieldLocation) reasonDetail.codeLocation();
-                try {
-                    final Class<?> type = Class.forName(location.typeName());
-                    try {
-                        final Field field = type.getDeclaredField(location.fieldName());
-                        final Type fieldType = field.getType();
-                        if (fieldType instanceof Class) {
-                            final Dotted fieldClassDotted = Dotted.fromClass((Class<?>) fieldType);
-                            for (final Dotted allowedClass : classNames) {
-                                if (fieldClassDotted.equals(allowedClass)) {
-                                    return true;
-                                }
-                            }
-                        }
-                    } catch (final NoSuchFieldException | SecurityException e) {
+                final Dotted fieldType = location.fieldType();
+                for (final Dotted allowedClass : classNames) {
+                    if (fieldType.equals(allowedClass)) {
+                        return true;
```

Comment:
Does `return allowedClass.contains(fieldType);` work here?

Reply:
I would need to change `org.mutabilitydetector.unittesting.matchers.reasons.ProvidedOtherClass.dottedClassNames` to a `Collection`; it's an `Iterable` right now. It looks like I can do that without impacting the public API so I'll make that change.

Reply:
:thumbsup: